packages/core/src/modules/credentials/formats/jsonld/JsonLdCredentialFormatService.ts

```diff
@@ -330,41 +315,49 @@ export class JsonLdCredentialFormatService implements CredentialFormatService<Js
     })
   }
 
-  private compareCredentialSubject(
+  private verifyReceivedCredentialMatchesRequest(
     credential: W3cVerifiableCredential,
-    requestAsJson: JsonLdSignCredentialFormat
+    request: JsonLdFormatDataCredentialDetail
   ): void {
-    const request = JsonTransformer.fromJSON(requestAsJson.credential, W3cCredential)
+    // const request = JsonTransformer.fromJSON(requestAsJson.credential, W3cCredential)
 
-    if (!objectEquality(credential.credentialSubject, request.credentialSubject)) {
-      throw new AriesFrameworkError('Received credential subject does not match subject from credential request')
+    const jsonCredential = JsonTransformer.toJSON(credential)
+    delete jsonCredential.proof
+
+    // Check whether the received credential (minus the proof) matches the credential request
+    if (!areObjectsEqual(jsonCredential, request.credential)) {
+      throw new AriesFrameworkError('Received credential does not match credential request')
     }
-  }
 
-  private compareProofs(credentialProof: LinkedDataProof, requestAsJson: JsonLdSignCredentialFormat): void {
-    const request = JsonTransformer.fromJSON(requestAsJson, JsonLdCredentialDetail)
+    if (Array.isArray(credential.proof)) {
+      throw new AriesFrameworkError('Credential proof arrays are not supported')
+    }
 
-    const requestProof = request.options
+    if (request.options.created && credential.proof.created !== request.options.created) {
+      throw new AriesFrameworkError('Received credential proof created does not match created from credential request')
+    }
 
-    if (credentialProof.domain !== requestProof.domain) {
+    if (credential.proof.domain !== request.options.domain) {
```

Comment:
Can't we do a `areObjectsEqual` on credential.proof? Or do we want the explicit error messages?

Reply:
The right side of the equation are the request.options. Although overlapping with a lot of properties from the proof, it's not the same object (and thus will fail)

Reply:
Ah yeah so It would require a new function that checks if one is fully included in the other, might a bit overkill for here. Can see it as resolved.